packages/file_selector/file_selector_android/android/src/main/java/io/flutter/plugins/file_selector/FileSelectorPlugin.java

```diff
@@ -99,24 +104,41 @@ private void tearDown() {
     }
   }
 
+  @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
   @Override
   public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result rawResult) {
     if (activityState == null || activityState.getActivity() == null) {
       rawResult.error("no_activity", "file_selector plugin requires a foreground activity.", null);
       return;
     }
+
+    final HashMap arguments = (HashMap) call.arguments;
+
     MethodChannel.Result result = new MethodResultWrapper(rawResult);
     FileSelectorDelegate delegate = activityState.getDelegate();
+    delegate.clearCache();
     switch (call.method) {
       case METHOD_GET_DIRECTORY_PATH:
         delegate.getDirectoryPath(call, result);
         break;
       case METHOD_GET_SAVE_PATH:
         throw new UnsupportedOperationException("getSavePath is not supported yet");
       case METHOD_OPEN_FILE:
-        throw new UnsupportedOperationException("openFile is not supported yet");
+        String[] mimeTypes = getMimeTypes(arguments);
+        boolean isMultipleSelection = (boolean) arguments.get("multiple");
```

Comment:
adding to this, we should manage the arguments in the delegate's method, e.g. `delegate.openFile(call, result)`; and then having those arguments mapped there.

Also, please check how we're dealing with the arguments names, since they're constants mapping to the same in the dart code. I saw these added by you in the delegate, so we can reuse them there

Reply:
Arguments logic moved to delegate. Thanks!